src/antelope/stores/rex.ts

```diff
@@ -5,28 +5,161 @@
+        },
+        async getEscrowContractInstance(label: string) {
+            const address = (useChainStore().getChain(label).settings as EVMChainSettings).getEscrowContractAddress();
+            return this.getContractInstance(label, address);
+        },
+        async updateTotalStaking(label: string) {
+            this.trace('updateTotalStaking', label);
+            const contract = await this.getStakedSystemContractInstance(label);
+            const totalStaking = await contract.totalAssets();
+            this.setTotalStaking(label, totalStaking);
+        },
+        async updateRexDataForAccount(label: string, account: AccountModel | null) {
+            this.trace('updateRexDataForAccount', label, account);
+            useFeedbackStore().setLoading('updateRexDataForAccount');
+            try {
+                await Promise.all([
+                    this.updateWithdrawable(label),
+                    this.updateDeposits(label),
+                    this.updateBalance(label),
+                    this.updateTotalStaking(label),
+                ]);
+            } catch (error) {
+                console.error(error);
+                useFeedbackStore().unsetLoading('updateRexDataForAccount');
+            }
+        },
+        async updateWithdrawable(label: string) {
+            this.trace('updateWithdrawable', label);
+            const contract = await this.getEscrowContractInstance(label);
+            const address = useAccountStore().getAccount(label).account;
+            const withdrawable = await contract.maxWithdraw(address);
+            this.setWithdrawable(label, withdrawable);
+        },
+        async updateDeposits(label: string) {
+            this.trace('updateDeposits', label);
+            const contract = await this.getEscrowContractInstance(label);
+            const address = useAccountStore().getAccount(label).account;
+            const deposits = await contract.depositsOf(address);
+            this.setDeposits(label, deposits);
+        },
+        async updateBalance(label: string) {
+            this.trace('updateBalance', label);
+            const contract = await this.getEscrowContractInstance(label);
+            const address = useAccountStore().getAccount(label).account;
+            const balance = await contract.balanceOf(address);
+            this.setBalance(label, balance);
+        },
+        // commits ---------------
+        setWithdrawable(label: string, withdrawable: ethers.BigNumber) {
+            const num = parseFloat(ethers.utils.formatUnits(withdrawable, 18));
+            this.trace('setWithdrawable', label, num);
+            this.__rexData[label] = this.__rexData[label] || {};
+            (this.__rexData[label] as EvmRexModel).withdrawable = withdrawable;
+        },
+        setDeposits(label: string, deposits: EvmRexDeposit[]) {
+            this.trace('setDeposits', label, ... deposits.map(d => parseFloat(ethers.utils.formatUnits(d.amount, 18))));
```

Comment:
systemtoken.decimals or WEI_PRECISION should be used here rather than 18

Reply:
I took this chance to convert all plain "18"s I found on the code for the corresponding variable/constant.